### PR TITLE
ISSUE#2: Optimize User Data Retrieval in Authentication Middleware 

### PR DESCRIPTION
**Pull request by syamjir**
Hi@mohitisimmortal,
This pull request done with following changes:

Updated authenticateToken Middleware:
Attached the complete user data to req.user, eliminating the need for subsequent database queries in certain routes.

Updated getUserProfile Route Handler:
Removed the redundant database query to fetch the user profile, relying on the full user data already available in req.user.

Please review the changes and give feedback

Thank you!

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -1,241 +1,252 @@
-const bcrypt = require('bcrypt');
-const jwt = require('jsonwebtoken');
-const userSchema = require('../models/userSchema');
+const bcrypt = require("bcrypt");
```
Comment (mohitisimmortal) [inline backend/controllers/userController.js:1]: what the need for change single quote to double

**Comment by mohitisimmortal** [comment]
Thanks so much to solve main issue, but why did you change single quote to double quotes in require statements, also one more thing you format code due to which i have to see much code changes(very hectic).

**Comment by syamjir** [comment]
@mohitisimmortal ,You're very welcome! The change to double quotes and code formatting was likely due to the automatic formatting settings in the editor or linter, which standardizes quote styles and formatting. I understand the extra changes can be a hassle to review.

**Comment by mohitisimmortal** [comment]
ok got it, thanks again!

**Pull request closed without merging**